components/plurals/src/provider.rs

```diff
@@ -368,115 +377,705 @@ pub enum PluralElementsKeysV1 {
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de, V> serde::Deserialize<'de> for Box<PluralElementsPackedULE<V>>
+where
+    V: VarULE + ?Sized,
+    Box<V>: serde::Deserialize<'de> + PartialEq + fmt::Debug,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            let plural_elements =
+                PluralElements::<(FourBitMetadata, Box<V>)>::deserialize(deserializer)?;
+            Ok(zerovec::ule::encode_varule_to_box(&plural_elements))
+        } else {
+            let bytes = <&[u8]>::deserialize(deserializer)?;
+            PluralElementsPackedULE::<V>::parse_byte_slice(bytes)
+                .map(|ule| ule.to_owned())
+                .map_err(serde::de::Error::custom)
+        }
+    }
+}
+
+#[cfg(feature = "datagen")]
+impl<V> serde::Serialize for PluralElementsPackedULE<V>
+where
+    V: PartialEq + serde::Serialize + VarULE + ?Sized,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            let plural_elements = self.to_plural_elements();
+            plural_elements.serialize(serializer)
+        } else {
+            serializer.serialize_bytes(self.as_byte_slice())
         }
-        .to_unaligned();
-        self.specials
-            .iter()
-            .filter_map(|ule| (ule.0 == category).then_some(&ule.1))
-            .next()
-            .unwrap_or(&*self.other)
     }
 }
 
 #[cfg(feature = "datagen")]
-impl<'a, 'b> From<PluralElements<'b, str>> for PluralElementsV1<'a> {
-    fn from(value: PluralElements<'b, str>) -> Self {
+impl<'a, V> databake::Bake for &'a PluralElementsPackedULE<V>
+where
+    &'a V: databake::Bake,
+    V: VarULE + ?Sized,
+{
+    fn bake(&self, ctx: &databake::CrateEnv) -> databake::TokenStream {
+        ctx.insert("icu_plurals");
+        let bytes = (&self.bytes).bake(ctx);
+        databake::quote! {
+            // Safety: the bytes came directly from self.bytes on the previous line.
+            unsafe { icu_plurals::provider::PluralElementsPackedULE::from_byte_slice_unchecked(#bytes) }
+        }
+    }
+}
+
+#[cfg(feature = "datagen")]
+impl<'a, V> databake::BakeSize for &'a PluralElementsPackedULE<V>
+where
+    &'a V: databake::Bake,
+    V: VarULE + ?Sized,
+{
+    fn borrows_size(&self) -> usize {
+        self.bytes.len()
+    }
+}
+
+/// Helper function to property deserialize a `Cow<PluralElementsPackedULE<V>>`
+///
+/// Due to <https://github.com/rust-lang/rust/issues/130180>, you may need to qualify
+/// `V` when invoking this, like so:
+///
+/// `#[serde(deserialize_with = "deserialize_plural_elements_packed_cow::<_, str>")]`
+///
+/// See <https://github.com/unicode-org/icu4x/pull/1556>
+#[cfg(feature = "serde")]
+pub fn deserialize_plural_elements_packed_cow<'de, 'data, D, V>(
```

Comment:
this can be private, if you're using a cow you might as well use the `PluralElementsPackedCowStr` struct

Reply:
I left it public because if you want to deserialize something other than a `str`, you'll need it. I consider it part of the value proposition of `PatternElementsPackedULE`.

Reply:
I managed to make `PluralElementsPackedCow` be generic, so now I am able to make the fn be private.